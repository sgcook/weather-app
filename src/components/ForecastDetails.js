import React from "react";
import PropTypes from "prop-types";
import "../styles/ForecastDetails.css";

function ForecastDetails({ forecast }) {
  const { date, humidity, temperature, wind } = forecast;

  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details">
      <div className="forecast-summary__date">{formattedDate}</div>
      <div className="forecast-summary__max-temperature">
        Max: {temperature.max}&deg;C
      </div>
      <div className="forecast-summary__min-temperature">
        Min: {temperature.min}&deg;C
      </div>
      <div className="forecast-summary__humidity">Humidity: {humidity}</div>
      <div className="forecast-summary__wind_speed">Speed: {wind.speed}</div>
      <div className="forecast-summary__wind_direction">
        Direction: {wind.direction}
      </div>
    </div>
  );
}

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }),
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }),
    humidity: PropTypes.number,
    description: PropTypes.string,
  }).isRequired,
};

export default ForecastDetails;
