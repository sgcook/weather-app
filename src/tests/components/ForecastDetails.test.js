import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("Forecast Details", () => {
  const validProps = {
    forecast: {
      date: 1111111,
      temperature: {
        min: 12,
        max: 22,
      },
      wind: {
        speed: 13,
        direction: "s",
      },
      humidity: 30,
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastDetails forecast={validProps.forecast} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
