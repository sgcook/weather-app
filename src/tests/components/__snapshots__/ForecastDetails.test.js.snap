// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Details renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-summary__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-summary__max-temperature"
    >
      Max: 22°C
    </div>
    <div
      class="forecast-summary__min-temperature"
    >
      Min: 12°C
    </div>
    <div
      class="forecast-summary__humidity"
    >
      Humidity: 30
    </div>
    <div
      class="forecast-summary__wind_speed"
    >
      Speed: 13
    </div>
    <div
      class="forecast-summary__wind_direction"
    >
      Direction: s
    </div>
  </div>
</DocumentFragment>
`;
